src/shared/utils/flattenChildren.js

```diff
@@ -22,7 +23,12 @@ var warning = require('warning');
  * @param {!string} name String name of key path to child.
  * @param {number=} selfDebugID Optional debugID of the current internal instance.
  */
-function flattenSingleChildIntoContext(traverseContext, child, name, selfDebugID) {
+function flattenSingleChildIntoContext(
+  traverseContext: any,
```

Comment:
Can you try `mixed` instead of `any`. Any will ignore it whereas mixed will ensure that the type stays consistent thorough. It's usually better when you don't know the type.